connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -505,7 +505,8 @@ private WorkerTask buildWorkerTask(ClusterConfigState configState,
                     time, retryWithToleranceOperator);
         } else if (task instanceof SinkTask) {
             TransformationChain<SinkRecord> transformationChain = new TransformationChain<>(connConfig.<SinkRecord>transformations(), retryWithToleranceOperator);
-            retryWithToleranceOperator.reporters(sinkTaskReporters(id, connConfig, errorHandlingMetrics));
+            SinkConnectorConfig sinkConfig = new SinkConnectorConfig(plugins, connConfig.originalsStrings());
```

Comment:
connConfig should already be an instance of SinkConnectorConfig in this condition. Wondering if this needs to be reconstructed here again.

Reply:
Thanks, @mageshn! we need the dlq topic name from the sink connector config. Hence, we definitely have to create this object here? re-using the `connConfig` would miss out on those properties. Hope I'm not missing anything.

Reply:
Right. The method that calls this instantiates `ConnectorConfig`, not `SinkConnectorConfig`, so it needs to be constructed here.

Reply:
@rhauch That's right. I missed that. I assumed, the right instance would be constructed in the first place :)